indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -249,13 +252,27 @@ def _process(self):
         return jsonify(booking_id=booking_id, room_id=room_id)
 
 
-class RHBookingEventData(RHBookingBase):
+class RHLinkedObjectData(RHRoomBookingBase):
+    """Fetch data from event, contribution or session block"""
+    def _process_args(self):
+        type = request.view_args['type']
+        id = request.view_args['id']
+        self.linked_object = {
+            'event': lambda x: Event.get_one(x),
+            'contrib': lambda x: Contribution.get_one(x),
+            'session_block': lambda x: SessionBlock.get_one(x)
+        }[type](id)
+
     def _process(self):
-        if self.booking.event is None:
+        if self.linked_object is None:
             raise NotFound
-        if not self.booking.event.can_access(session.user):
+        if isinstance(self.linked_object, Event):
+            event = self.linked_object
+        else:
+            event = self.linked_object.event
+        if not event.can_access(session.user):
```

Comment:
Why not check access on the linked object? An event you can access might have protected contributions!